File: zero_task_java/src/include/Result.java
package include;

public class Result {
    public String expression;
    public Result left;
    public Result right;
    public int number = 0;

    public Result(String expression) {
        this.expression = expression;
    }

    public Result(String expression, Result a) {
        this.expression = expression;
        left = a;
        number = 1;
    }

    public Result(String expression, Result a, Result b) {
        this.expression = expression;
        left = a;
        right = b;
        number = 2;
    }
}

File: ThirdTaskKotlin/src/include/Result.kt
package include

class Result {
    var expression: String
    var left: Result? = null
    var right: Result? = null
    var number = 0

    constructor(expression: String) {
        this.expression = expression
    }

    constructor(expression: String, a: Result) {
        this.expression = expression
        left = a
        number = 1
    }

    constructor(expression: String, a: Result, b: Result) {
        this.expression = expression
        left = a
        right = b
        number = 2
    }

    constructor(expression: String, left: Result?, right: Result?, number: Int) {
        this.expression = expression
        this.left = left?.copy()
        this.right = right?.copy()
        this.number = number
    }

    fun copy() = Result(expression, left, right, number)
}